### PR TITLE
Azure Monitor: support AzureCredentials in common format on backend

### LINKED ISSUE #60782: Azure Monitor: Update backend to common AzureCredentials
For unification of the Azure authentication and support for new authentication types, Azure Monitor needs to be migrated to the common credentials format.

The existing credentials format may still be supported by backend as legacy to not break the existing datasources.

See related work in Azure Data Explorer datasource:
- https://github.com/grafana/azure-data-explorer-datasource/issues/360
- https://github.com/grafana/azure-data-explorer-datasource/pull/459

### PR DESCRIPTION
**Pull request by kostrse**
**What is this feature?**

This PR makes backend to understand stored Azure credentials in both legacy and the new "common" credentials format.
It allows to update the frontend code to save newly saved credentials in common format, rather than continuing to use the legacy format.

**Why do we need this feature?**

We need it to unify the Azure authentication logic across all the datasources and reuse as much frontend and backend code as possible.
The user identity authentication (Current User) will support only the new credentials format once introduced.

**Who is this feature for?**

It is internal refactoring of credentials format which should not affect user experience.
This change will not be visible to end users.

One benefit end users will have is that in case of automation scenarios (e.g. Terraform), they could use the same datasource configuration schema for all Azure datasources, without having to adopt their automation scripts to specifics of particular datasource.

**Which issue(s) does this PR fix?**:

Fixes #60782

**Special notes for your reviewer:**

This PR was influenced by prior work done for ADX datasource:
* https://github.com/grafana/azure-data-explorer-datasource/pull/459

### CODE REVIEW

File: pkg/tsdb/azuremonitor/azmoncredentials/builder_test.go [hunk 1]
```diff
@@ -0,0 +1,215 @@
...
+			assert.IsType(t, &azcredentials.AzureClientSecretCredentials{}, result)
+			credential := (result).(*azcredentials.AzureClientSecretCredentials)
+
+			assert.Equal(t, azsettings.AzureChina, credential.AzureCloud)
+			assert.Equal(t, "LEGACY-TENANT-ID", credential.TenantId)
+			assert.Equal(t, "LEGACY-CLIENT-ID", credential.ClientId)
+			assert.Equal(t, "FAKE-LEGACY-SECRET", credential.ClientSecret)
+		})
+
+		t.Run("should return client secret credentials when auth type is specified but configuration present", func(t *testing.T) {
```
Comment (aangelisc) [inline pkg/tsdb/azuremonitor/azmoncredentials/builder_test.go:80]:
```suggestion
		t.Run("should return client secret credentials when auth type is not specified but configuration present", func(t *testing.T) {
```

**Review by aangelisc — APPROVED** [review]
One nit, otherwise LGTM. Thanks @kostrse!

**Comment by aangelisc** [comment]
I think you'll have to merge main into this branch so that CI can pass

**Comment by kostrse** [comment]
Hi @aangelisc, I resolved the merge conflicts but please hold on this PR for a bit.
I see strage behavior of MSI request in `Azure/azure-sdk-for-go`, not related my changes, but I want to confirm that everything works.

**Comment by kostrse** [comment]
@aangelisc It was a problem in my test setup, MSI works fine. It can be merged.

**Pull request merged**